redpanda/resources/cluster/resource_cluster.go

```diff
@@ -263,7 +303,26 @@ func (c *Cluster) Read(ctx context.Context, req resource.ReadRequest, resp *reso
 		NetworkID:       types.StringValue(cluster.NetworkId),
 		ID:              types.StringValue(cluster.Id),
 		ClusterAPIURL:   types.StringValue(clusterURL),
-	})...)
+	}
+	if cluster.GetAwsPrivateLink() != nil {
+		if cluster.GetAwsPrivateLink().GetEnabled() {
+			pl, dg := awsPrivateLinkStructToModel(ctx, cluster.GetAwsPrivateLink())
+			if dg.HasError() {
+				resp.Diagnostics.Append(dg...)
+			}
+			persist.AwsPrivateLink = pl
+		}
```

Comment:
If Private Link isn't enabled the related data is non-actionable and should not be persisted IMO. 

I gated it like this to avoid panics

Reply:
Yes, I agree that the data should not perform any action if is not `enabled`... but is the API that should decide that, we should store whatever value they pass, otherwise we could have errors on reconciliation.

For example, right now if you run an apply with a list on `AwsPrivateLink.allowed_principals` but `enabled false` you'll get a successful apply and save this to state:

```

  # redpanda_cluster.test will be created
  + resource "redpanda_cluster" "test" {
      + allow_deletion    = true
      + aws_private_link  = {
          + allowed_principals = [
              + "foo",
              + "bar",
            ]
          + enabled            = false
        }
```

But I think (I'm testing it but it takes some time to create a cluster haha), that when you read the state back, you won't fill the aws_private_link and there will be a mismatch in the state and we will end up getting something like:

```
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to redpanda_cluster.test ....
```

Reply:
Tests finished, and I got:

```
redpanda_cluster.test: Still creating... [37m21s elapsed]
redpanda_cluster.test: Still creating... [37m31s elapsed]
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to redpanda_cluster.test, provider "provider[\"registry.terraform.io/redpanda-data/redpanda\"]" produced an
│ unexpected new value: .aws_private_link: was
│ cty.ObjectVal(map[string]cty.Value{"allowed_principals":cty.ListVal([]cty.Value{cty.StringVal("foo"), cty.StringVal("bar")}),
│ "enabled":cty.False}), but now null.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

Reply:
Error was related to nil issues, fixed